nist_test.c

```diff
@@ -0,0 +1,243 @@
+#include "api.h"
+#include "nist_params.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <string.h>
+#include "xmss.h"
+
+#define XMSS_SIGNATURES 64
+
+unsigned long long t[XMSS_SIGNATURES];
+
+#if DEBUG
+static void print_hex(const unsigned char *a, int length, const char *string)
+{
+    printf("%s[%d] = \n", string, length);
+    for (int i = 0; i < length; i++)
+    {
+        printf("%02x", a[i]);
+    }
+    printf("\n");
+}
+#endif
+
+static int cmp_llu(const void *a, const void *b)
+{
+    if (*(unsigned long long *)a < *(unsigned long long *)b)
+        return -1;
+    if (*(unsigned long long *)a > *(unsigned long long *)b)
+        return 1;
+    return 0;
+}
+
+static unsigned long long median(unsigned long long *l, size_t llen)
+{
+    qsort(l, llen, sizeof(unsigned long long), cmp_llu);
+
+    if (llen % 2)
+        return l[llen / 2];
+    else
+        return (l[llen / 2 - 1] + l[llen / 2]) / 2;
+}
+
+static unsigned long long average(unsigned long long *t, size_t tlen)
+{
+    unsigned long long acc = 0;
+    size_t i;
+    for (i = 0; i < tlen; i++)
+    {
+        acc += t[i];
+    }
+    return acc / (tlen);
+}
+
+static void print_results(unsigned long long *t, size_t tlen)
+{
+    size_t i;
+    for (i = 0; i < tlen - 1; i++)
+    {
+        t[i] = t[i + 1] - t[i];
```

Comment:
Not clear from the function name why you need to do this operation. What's in the input array `t`?